app/app.go

```diff
@@ -491,11 +490,23 @@ func wireValidatorMock(conf Config, pubshares []eth2p0.BLSPubKey, sched core.Sch
 				return
 			}
 
-			err = validatormock.Attest(ctx, cl.(*eth2http.Service), signer, eth2p0.Slot(duty.Slot), pubshares...)
-			if err != nil {
-				log.Warn(ctx, "Attestation failed", z.Err(err))
-			} else {
-				log.Info(ctx, "Attestation success", z.I64("slot", duty.Slot))
+			switch duty.Type {
```

Comment:
@dB2510 this nested function is very long, perhaps extract it to its own function that returns the actual function would make it more readable

Reply:
Sure